src/lib/mariaDB.js

```diff
@@ -0,0 +1,374 @@
+      // as objects are passed by reference
+      columns = columns.concat(['slug']);
+    }
+    database.select(...columns)
+    .from('authors')
+    .whereIn('slug', slugs)
+    .then((rows) => {
+      database.destroy();
+      resolve(rows);
+    })
+    .catch((e) => {
+      console.log("error");
+      database.destroy();
+      throw new Error(e);
+    })
+  })
+}
+
+export function dbAuthorArticleQuery(slugs) {
+  // slugs function parameter is an array of author slugs
+  // of which to fetch the articles they have written.
+  // The function returns an object with author slugs
+  // as keys and values being arrays of article slugs
+  // sorted by most recent article first.
+  return new Promise((resolve, reject) => {
+    const database = knex(knexConnectionObject);
+    database.select('posts.slug as post_slug', 'authors.slug as author_slug')
+    .from('authors')
+    .innerJoin('authors_posts', 'authors.id', '=', 'author_id')
+    .innerJoin('posts', 'posts.id', '=', 'post_id')
+    .innerJoin('posts_meta', 'posts.id', '=', 'posts_meta.id')
+    .whereIn('authors.slug', slugs).orderBy('gazelle_published_at', 'desc')
+    .then((rows) => {
+      // rows is an array of objects with keys author_slug and post_slug
+      const data = {};
+      rows.forEach((row) => {
+        // This will input them in chronological order as
+        // the query was structured as so.
+        if (!data.hasOwnProperty(row.author_slug)) {
+          data[row.author_slug] = [row.post_slug];
+        }
+        else {
+          data[row.author_slug].push(row.post_slug);
+        }
+      });
+      database.destroy();
+      resolve(data);
+    })
+    .catch((e) => {
+      database.destroy();
+      throw new Error(e);
+    })
+  })
+}
+
+export function dbInfoPagesQuery(slugs, columns) {
+  // parameters are both expected to be arrays
+  // first one with page slugs to fetch
+  // second one which columns to fetch from the db
+  return new Promise((resolve, reject) => {
+    const database = knex(knexConnectionObject);
+    // So the Falcor Router knows which author we're talking about
+    if (columns.find((col) => {return col === "slug"}) === undefined) {
+      // Use concat to make a copy, if you just push
+      // it will change pathSet in the falcorPath
+      // as objects are passed by reference
+      columns = columns.concat(['slug']);
+    }
+    database.select(...columns)
+    .from('info_pages')
+    .whereIn('slug', slugs)
+    .then((rows) => {
+      database.destroy();
+      resolve(rows);
+    })
+    .catch((e) => {
+      database.destroy();
+      throw new Error(e);
+    })
+  })
+}
+
+export function dbArticleQuery(slugs, columns) {
+  // parameters are both expected to be arrays
+  // first one with article slugs to fetch
+  // second one which columns to fetch from the posts_meta table
+  return new Promise((resolve, reject) => {
+    const database = knex(knexConnectionObject);
```

Comment:
Yeah I've been thinking about that as well.

I googled a bit and it seems like for web applications where many users will be using it maybe it's a good idea.

But then also something I wanted to run by you was, all this code will ALWAYS only run on the server right?

Because if that's so, then I guess it could be okay to just keep 1 connection open right? since it's only the server that'll be connecting.

But to be honest I also thought about this and am a bit confused about what exact best practice would be, because I was also aware there'll be quite a big overhead.